File: code/771.宝石与石头.go
/*
 * @lc app=leetcode.cn id=771 lang=golang
 *
 * [771] 宝石与石头
 */

// @lc code=start
func numJewelsInStones(J string, S string) int {
	m := make(map[rune]int)
	for _, c := range J {
		m[c]++
	}

	count := 0
	for _, c := range S {
		if _, ok := m[c]; ok {
			count++
		}
	}
	return count
}

// @lc code=end

File: code/771.宝石与石头.rs
/*
 * @lc app=leetcode.cn id=771 lang=rust
 *
 * [771] 宝石与石头
 */

// @lc code=start
use std::collections::HashMap;
impl Solution {
    pub fn num_jewels_in_stones(j: String, s: String) -> i32 {
        let mut map = HashMap::new();
        for c in j.chars() {
            map.insert(c, 1);
        }
        let mut count: i32 = 0;
        for c in s.chars() {
            match map.get(&(c)) {
                Some(n) => {
                    count = count + 1
                }
                None => {}
            }
        }
        count
    }
}
// @lc code=end

